Bottom: 5df38d6d436ede264ffdcd2bae144e5d5c10186b
Top:    61f8c5dbe4fbd6855b3d99f1698a996df955cdd9
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-09 19:18:38 +0000

Refresh of add-polygon-parentage-tree

---

diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index 0277af5..316cb55 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -119,6 +119,8 @@ a zoom in/out.
 #include "gui-trackball.h"
 #include "snavi.h"
 
+#include "polyarea.h"
+
 #ifdef HAVE_LIBDMALLOC
 #include <dmalloc.h>
 #endif
@@ -1275,6 +1277,12 @@ ghid_polygon_debug_selection_changed_cb (GtkTreeSelection *treeselection, gpoint
   gtk_tree_model_get (model, &iter,
                       POLYGON_DEBUG_COLUMN_POLYAREA, &ghidgui->debugged_polyarea,
                       -1);
+
+  if (ghidgui->debugged_polyarea != NULL) {
+    fprintf (stderr, "Printing polyarea %p\n", ghidgui->debugged_polyarea);
+    poly_dump (ghidgui->debugged_polyarea);
+  }
+
   ghid_invalidate_all ();
 }
 
diff --git a/src/polyarea.h b/src/polyarea.h
index 637c497..86bf9c3 100644
--- a/src/polyarea.h
+++ b/src/polyarea.h
@@ -159,6 +159,8 @@ struct POLYAREA
     POLYPARENTAGE parentage;
 };
 
+void poly_dump (POLYAREA *p);
+
 BOOLp poly_M_Copy0(POLYAREA ** dst, const POLYAREA * srcfst);
 void poly_M_Incl(POLYAREA **list, POLYAREA *a);
 
diff --git a/src/polygon1.c b/src/polygon1.c
index d712158..9036984 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -159,7 +159,7 @@ pline_dump (VNODE * v)
   while ((v = NEXT_VERTEX(v)) != s);
 }
 
-static void
+/*static */void
 poly_dump (POLYAREA * p)
 {
   POLYAREA *f = p;
